miner/worker.go

```diff
@@ -198,6 +202,11 @@ func (miner *Miner) generateWork(genParam *generateParams, witness bool) *newPay
 		}
 	}
 
+	// OP-Stack addition: Jovian maxes the block.gasUsed with the calldata footprint
+	if miner.chainConfig.IsDAFootprintBlockLimit(work.header.Time) && work.daFootprint > work.header.GasUsed {
+		work.header.GasUsed = work.daFootprint
+	}
+
```

Comment:
See other PR, relevant comment: https://github.com/ethereum-optimism/op-geth/pull/655/files#r2391365500

The header modification should maybe just be part of the shared consensus-engine `Finalize` call (or `FinalizeAndAssemble`)

Also see `FinalizeAndAssemble` call in the code of `generateWork` here near the end.

Reply:
Yes I agree, I also wanted to do this, but then I thought I'm gonna remove the `gasUsed` update anyways when I switch to using the `blobGasUsed` field so I just left it as is.